sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -82,47 +97,76 @@ public ServiceBusMessage(byte[] body) {
      * @throws NullPointerException if {@code receivedMessage} is {@code null}.
      */
     public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage) {
-        this.body = receivedMessage.getBody();
+        Objects.requireNonNull(receivedMessage, "'receivedMessage' cannot be null.");
+
+        this.amqpAnnotatedMessage = new AmqpAnnotatedMessage(receivedMessage.getAmqpAnnotatedMessage());
         this.context = Context.NONE;
-        setMessageId(receivedMessage.getMessageId());
-        setScheduledEnqueueTime(receivedMessage.getScheduledEnqueueTime());
-        setContentType(receivedMessage.getContentType());
-        setCorrelationId(receivedMessage.getCorrelationId());
-        setLabel(receivedMessage.getLabel());
-        setPartitionKey(receivedMessage.getPartitionKey());
-        setReplyTo(receivedMessage.getReplyTo());
-        setReplyToSessionId(receivedMessage.getReplyToSessionId());
-        setTimeToLive(receivedMessage.getTimeToLive());
-        setTo(receivedMessage.getTo());
-        setSessionId(receivedMessage.getSessionId());
-        setViaPartitionKey(receivedMessage.getViaPartitionKey());
+
+        final AmqpDataBody amqpDataBody = ((AmqpDataBody) receivedMessage.getAmqpAnnotatedMessage().getBody());
```

Comment:
We should also make a copy of the annotated message rather than passing a reference to it or else they may change parts of the annotated message which affect other references of it.